starter_code/src/movies.js

```diff
@@ -1,20 +1,100 @@
 /* eslint no-restricted-globals: 'off' */
 // Turn duration of the movies from hours to minutes 
 
+function turnHoursToMinutes(arr) {
+    return arr.map(function(movie) {
+        var newMovie = Object.assign({}, movie);
+
+        var durationArray = newMovie.duration.split(" ");
+
+        if (durationArray.length === 2) {
+            var hours = durationArray[0].replace("h","");
+            hours = parseInt(hours) * 60;
+            var minutes = durationArray[1].replace("min","");
+            minutes = parseInt(minutes);
+
+            newMovie.duration = hours + minutes;  
+
+        } else if (durationArray[0].includes('h')) {
+            var hours = durationArray[0].replace("h","");
+            hours = parseInt(hours) * 60;
+            newMovie.duration = hours;
+
+        } else {
+            var minutes = durationArray[0].replace("min","");
+            minutes = parseInt(minutes);
+            newMovie.duration = minutes;   
+
+        }
+        return newMovie;
+    });
+}
 
 // Get the average of all rates with 2 decimals 
 
+function ratesAverage(movies) {
+    var rates = movies.reduce(function(acc, movie) {
+        return acc += parseFloat(movie.rate);
```

Comment:
El primer test que te da error se produce por ese `parseFloat`. La función `parseFloat` convierte un entero a decimal, pero no tiene capacidad de convertir `String` a `Number`. Qué problema puede dar eso? pues que en el test, Jasmine te procura en uno de los casos, una de las películas con `rate: " "`. Con lo que si intentas hacer `parseFloat` de una cadena vacía, te devuelve `NaN` (`Not a Number`). A partir de ahí ya no puedes sumar más puntajes de películas por ese `NaN` y Jasmine te dará error.

Solución!, ya que usas, al hacer la media, `toFixed(2)`, no necesitas hacer `parseFloat`. La clase `Number` tiene la capacidad de convertir una cadena potencialmente parseable a su igual numérico. Si no es parseable, devolverá `NaN`, para cuyo caso, un `string` vacío `Number` es capaz de parsearlo a un valor numérico falso, es decir, 0. A partir de ahí, Jasmine puede seguir operando. :cactus: